hw/ip/otbn/data/bignum-insns.yml

```diff
@@ -501,6 +502,7 @@
     result = a & b_shifted
 
     WDR[d] = result
+    update_mlz_flags(0, result)
```

Comment:
At least: Can you add an appropriate function to the pseudo-code description in the docs, and align the function naming with the other functions, e.g. `UpdateMLZFlags()`? 

Preferred: To make the behavior a bit more obvious when reading the docs and aligned with what we do e.g. for BN.ADD, I'd go for a less indirect approach, e.g.

```
flags_out = FlagsForResult(result)
FLAGS[0] = {M: flags_out.M, L: flags_out.L, Z: flags_out.Z}
```

Reply:
Hmm. The problem is that we don't clobber C, so
```
FLAGS[0] = {M: flags_out.M, L: flags_out.L, Z: flags_out.Z}
```
isn't quite right.

Any ideas for a neat syntax that captures this?

Reply:
Indeed. I'm not opposed to slightly longer syntax to make this behavior very clear, it's something that's easy to miss for a SW developer.

```
FLAGS[0].M = flags_out.M
FLAGS[0].L = flags_out.L
FLAGS[0].Z = flags_out.Z
```

or something along those lines (I think I'd go for that one)

```
prev_c_flag = FLAGS[0].C
FLAGS[0] = {M: flags_out.M, L: flags_out.L, Z: flags_out.Z, C: prev_c_flag}
```

Reply:
Well, the latter *is* the body of update_mlz_flags :-) I'll expand out the documentation example as you suggest, but leave the implementation unchanged. I'm hoping that eventually we'll be auto-generating the documentation from the implementation, at which point this will be a problem again, but let's punt on that until it's an issue.